Fix repo file upload timeout

mliudev opened the pull request:
Fixes the issue where `working` does not update when Dropzone is uploading. See commit messages for details on the fix.

mliudev committed:
Configure webpack to export files as name-s3commander.*

This makes it easier to identify which app these files came from.

mliudev committed:
Fix a comparator misuse.

!== checks if the values are equal as well as if they are the same type,
which is what we want here.

mliudev committed:
Fix indentation.

This block is indented further than it should be relative to the other
lines.

mliudev committed:
Fix updating the working state.

According to https://stackoverflow.com/a/14049482, a new scope is
created when elements are created with ng-if. This means that two way
binding does not update outer scope attributes from the innerscope. This
is because the inner scope creates its own property for primitives which
hides the parent scope property. We work around this by passing a
function instead of a property.

mliudev commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[general-229] Guanxi Chen needs an ability to upload 28G + (Opened)](https://trello.com/c/1iI3nHqp/3303-general-229-guanxi-chen-needs-an-ability-to-upload-28g-opened)

abannerjee commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Fix session timeout when uploading large files](https://trello.com/c/6QDQvMIY/3379-fix-session-timeout-when-uploading-large-files)

Commit:
Add function to get the full file path given a folder and file.

This function is added to simplify calculating the 'key' parameter
needed by several aws api calls. The aws 'key' is the full path of
a file on s3.

Commit:
Bump s3commander version to 0.8.0.

This version of s3commander supports multi-part uploading to S3.

Commit:
Update README to include ExposeHeader ETag in CORS policy.

In order to use S3 ManagedUpload to upload files to S3, the
S3 bucket needs to include ExposeHeader ETag in the CORS policy.

Commit:
Update AWS S3 getUploadConfig to not create pre-signed post url.

Since the aws sdk is used to upload files, creating a pre-signed
post url is no longer required. This function has been updated to
simply return the S3 bucket URL and bucket name.

Commit:
Add backend function to upload files using multi-part.

A new function is added to AmazonS3Backend to upload files
using AWS ManagedUpload. A new ManagedUpload object is
instantiated and attached to the File which is then used
send files to S3 as well as listen for upload progress.

Commit:
Add backend function to cancel uploading files.

A new function is added to AmazonS3Backend to cancel uploading a
file using ManagedUpload. The ManagedUpload abort() function returns
void therefore there is no data to return to the dropzone directive
so the backend function simply returns true.

Commit:
Override Dropzone uploadFiles method and define accept callback.

The uploadFiles function is overridden to allow uploading files
using the backend 'uploadFile' method instead of using
Dropzone to upload files.

The backend method will allow uploading files in chunks and
will support uploading very large files.

Commit:
Remove Dropzone 'sending' event handler.

This event handler no longer needs to be defined since Dropzone is
not used for sending files, and this handler never gets called.

In addition, the code which enable prompts when user attempts to
navigate away from the page during upload has been moved out of the
event handler and placed in the overridden Dropzone 'uploadFiles'
function.

Commit:
Remove backend function 'updateFormData'.

The updateFormData function was previously used to update the
formData by adding the 'key' field needed to make a valid POST to
AWS.

This function is no longer called since file uploads now go through
the backend uploadFile function which uses AWS ManagedUpload to
send the files.

Commit:
Define dropzone filesizeBase as 1024 to display correct file sizes.

Commit:
Set AWS ManagedUpload part size to 10MB with 1 concurrent upload.

This commit sets the managed upload part size to 10mb. If part sizes are
too large, the request times out and gets cancelled. Upload requests to
AWS timeout after 2 minutes. In addition, the concurrent uploads have
been reduced from the default of 4 to 1.

abannerjee committed:
Merge pull request #26 from nimbis/dropzone-multipart-rev3

Dropzone multipart rev3
